### ISSUE TITLE
Disable product translation

### ISSUE THREAD

**Issue by jansan20** [hyyan/woo-poly-integration#280]
#### Can you reproduce this issue on default Wordpress theme (eg Storefront)?
Yes
#### Can you reproduce this issue when all other plugins are disabled except WooCommerce, Polylang and Hyyan WooCommerce Polylang Integration?
Yes
#### What product versions and settings are you using when this issue occurs?
* PHP: 7.0.22-0ubuntu0.16.04.1
* WordPress: 4.9.1
* WooCommerce: 3.2.5
* Polylang: 2.2.7
* Hyyan WooCommerce Polylang Integration: 1.0.4
* Browser: Google Chrome Version 63.0.3239.84 (Official Build) (64-bit)

#### What Happened Instead

This is not an problem it is rather a question, because I do not seem to be able to find an answer online or in any forum.

SITUATION:
I use Woocommerce together with Symbiostock Pro 4.0.1 in order to sell downloadable Photos. This allows me to easily upload photos in bulk via FTP or whatever and creates products according to a preset beautifully.

HOWEVER:
These products are by default not assigned any language. So they do not show up anywhere immediately. If I assign a language (I can do this in bulk very quickly), they show up only under the particular language.

But now, I would need to duplicate them in bulk, and assign the next language to the duplicates in order to make them show up on the website, or upload the same photos for any language.

QUESTIONS:
1. How can I NOT translate products (disable product translations), so that they are treated like categories or tags (for which the translations can be disabled) in terms of translation? I want the same products to show up regardless of language. I do not have different descriptions, categories, or anything that would require translating, only menus and Woocommerce pages.
2. If number 1 is not easily done, how can I duplicate products in bulk, so that everything is exactly the same, in order to later assign a second language? Duplication plugins leave out parts of the tags, or they do not bulk cloning.

If no solution to this problem can be found I would need to create a SQL query to achieve what I need, but I was hoping for some more elegant solution. 

I assume there may be other situations where you would not need/want to have translations for products.

Thank you very much in advance for any help.

**Comment by hyyan** [comment]
@jansan20 In your case you do not need woo-poly-integration plugin at all, the plugin main job is to translate products, if you do not need product translation then you do not need the plugin

**Comment by jansan20** [comment]
Hi hyyan!

It is an honour to hear from you directly! Thanks for the quick response.

I tried removing the plugin (not only disabling), but I can still see the flags and translations in the woocommerce product manager. I also tried removing all products and readding them. They still show up only in one language. 

I do not have any caching on at the moment.

Any idea?

**Comment by Jon007** [comment]
Quick answer - uninstall Polylang.
Polylang assumes and requires that there will be different copies of each post for each language and hooks all the wordpress queries to add a language filter.

Otherwise if you are still planning to use Polylang for other types of post there are two possibilities:
1. go to Languages, Settings, Custom post types and Taxonomies and clear the checkboxes for Products and anything else related to wooCommerce
2. use wordpress multisite and run your wooCommerce on a subsite which does not have Polylang installed.  If it makes sense in your context this is also the fastest solution as each site gets its own tables you don't have wooCommmerce loaded on your non-shop pages and you don't have Polylang loaded on your WooCommerce pages.

**Comment by jansan20** [comment]
Hi, I know some time has passed. But since this thread showed up upon my Googling of the same problem, I decided to share my own solution.

I need everything to be translated except the products (photos) themselves. The customer does not want to upload these 3 times (number of languages) and they don't want to duplicate them manually.

After some research, I found out that:

1. Every post(product) has an entry in the wp_term_relationships table in the database. This determines among others the language affiliation.

2. I determined my term_taxonomy_id values for each language (if you don't want to search around the database, just assign one product to each language individually and look for changes in wp_term_relationships).

3. Once I knew the term_taxonomy_id values for my languages, I exported all my existing product ids:
SELECT ID FROM `wp_posts` where post_type = 'product';

4. And gave them all the necessary language affiliations (737 was my id for Spanish, I ran the following query three times, with the corresponding term_taxonomy_ids, to account for each of the languages):

INSERT IGNORE INTO `wp_term_relationships` (`object_id`, `term_taxonomy_id`, `term_order`) VALUES 
("16621", "737", "0"),
("16622", "737", "0"),
...
("19321", "737", "0");

5. At the end I added a trigger to my database, to add each language affiliation, every time a product gets created:

CREATE TRIGGER `langfix` AFTER INSERT ON `wp_posts`
 FOR EACH ROW IF (NEW.post_type = 'product') THEN
INSERT IGNORE INTO `wp_term_relationships` VALUES (NEW.`id`, 23, 0), (NEW.`id`, 26, 0),(NEW.`id`, 737, 0);
END IF

Now, every time the customer uploads a photo to Symbiostock, it becomes a product of all three languages at the same time and shows up on each site language version.

Let me know should you have a better solution.

Thanks.
